lib/honeycomb/propagation/aws.rb

```diff
@@ -47,20 +47,46 @@ def get_fields(fields)
 
     # Serialize trace headers
     module MarshalTraceContext
-      def to_trace_header
+      def to_trace_header(propagation_context: nil)
+        if propagation_context.nil?
```

Comment:
This feels wrong. We should ALWAYS have a propagation context - this assembly of data that SHOULD be in a propagation context from wherever it's coming from (globals, ???) is awkward - this means that this AWS implementation needs to know about our internals. I thought the general idea was that these hooks ONLY needed to know about propagation contexts.

Propagation context should be a required argument, and the function that calls `to_trace_header` should be requiring it.

If `to_trace_header` is a legacy API, then we should leave it alone and deprecate it - having it fill in a propagation context and then call the new propagation-context hook.

Reply:
Ah, saw your comment further down. This just generally applies to all `to_trace_header` implementations. In general for this sort of refactoring, I am strongly for reimplementing legacy interfaces in terms of new, cleaner, interfaces, rather than trying to accommodate both use cases with a single code path.

Reply:
@dsoo Took a stab at refactoring this, let me know what you think: https://github.com/honeycombio/beeline-ruby/pull/113/commits/52472b27d8e4b2311ec9c82992c0160480a9455a